src/core/probe/bpf/include/common_defs.h

```diff
@@ -32,6 +31,52 @@ static __always_inline bool collection_enabled() {
 #define COMMON_SECTION_CORE	0
 #define COMMON_SECTION_TASK	1
 
+/* Current log level. */
+const volatile u8 log_level = 0;
+
+/* Aligned with the log crate. */
+enum {
+	LOG_ERROR = 1,
+	LOG_WARN,
+	LOG_INFO,
+	LOG_DEBUG,
+	LOG_TRACE,
+};
+
+/* Checks if the level will be reported. */
+#define retis_log_enabled(__lvl)			\
+({							\
+	bool active = false;				\
+	if (log_level && __lvl <= log_level)		\
+		active = true;				\
+	active;						\
+})
+
+/* Log macros must be used carefully and preferrably in the
+ * {error,slow} path.
+ * Useful exceptions must use a high log level (ideally LOG_TRACE).
+ */
+#define retis_log(lvl, fmt, args...)					\
+({									\
+	struct retis_log_event *__log = NULL;				\
+	if (retis_log_enabled(lvl)) {					\
+		__log = bpf_ringbuf_reserve(&log_map, sizeof(struct retis_log_event), 0); \
+		if (__log) {						\
+			__builtin_memset(__log, 0, sizeof(struct retis_log_event)); \
```

Comment:
We might be able to remove this, as `__log->level` and `log->data` are set below. BPF_SNPRINTF sets a `\0` at the end of the formatted string (or at the end of the buffer if the formatted str size equals or is above it).

Reply:
uhm, that's right. I think I saw weird things w/o the `memset()` because of the improper cstr representation using `.trim_end_matches()`